app/models/recurring_donation.rb

```diff
@@ -79,6 +79,15 @@ def will_attempt_again?
     !failed? && !cancelled? && (end_date.nil? || end_date > Time.current);
   end
 
+  def cancel!(email)
+    if persisted?
+      self.active = false
+      self.cancelled_by = email
+      self.cancelled_at = Time.current
+      save!
+    end
```

Comment:
What's the reason for checking if it's persisted in this case?

Reply:
My thinking was that if it was not yet saved, it might be weird to run cancel on something that, effectively, doesn't exist yet. That said, now that I think about it, I think we should remove the persisted check.

Reply:
I've removed the persisted check.

Reply:
Gotcha. I think it makes sense, but maybe we should find a way to ensure that it never gets to this point, it's weird that a recurring donation could be canceled by someone without ever being saved if it's something we worry about.